lib/src/Control/Arrow/Environment.hs

```diff
@@ -1,13 +1,14 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE FunctionalDependencies #-}
 {-# LANGUAGE Arrows #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 module Control.Arrow.Environment where
 
 import Control.Arrow
 import Control.Arrow.Utils
 
 class Arrow c => ArrowEnv x y env c | c -> x, c -> y, c -> env where
-  lookup :: c x y
+  lookup :: c x (Maybe y)
```

Comment:
Please leave the interface for existing type classes unchanged if possible, first because you are going to break a lot of code that uses this type class and second because it does not work for some abstract domains.